source/MaterialXTest/ShaderValid.cpp

```diff
@@ -402,6 +404,78 @@ class ShaderValidTestOptions
     MaterialX::FilePath glslShaderGeometry = "shaderball.obj";
 };
 
+// Per language profile times
+class LanguageProfileTimes
+{
+public:
+    void print(const std::string& label, std::ostream& output) const
+    {
+        output << label << std::endl; 
+        output << "\tGeneration: " << generationTime << " seconds" << std::endl;;
+        output << "\tCompile: " << compileTime << " seconds" << std::endl;
+        output << "\tRender: " << renderTime << " seconds" << std::endl;
+        output << "\tI/O: " << ioTime << " seconds" << std::endl;
+        output << "\tImage save: " << imageSaveTime << " seconds" << std::endl;
+    }
+    double totalTime() const
+    {
+        return generationTime + compileTime + renderTime + ioTime;
+    }
+    double generationTime = 0.0;
+    double compileTime = 0.0;
+    double renderTime = 0.0;
+    double ioTime = 0.0;
+    double imageSaveTime = 0.0;
+};
+
+//
+// Shader validation profiling structure
+//
+class ShaderValidProfileTimes
+{
+public:
+    double totalTime() const
+    {
+        return setupTime + glslTimes.totalTime() + oslTimes.totalTime();
+    }
+
+    LanguageProfileTimes glslTimes;
+    LanguageProfileTimes oslTimes;
+    double setupTime = 0;
+    unsigned int elementsTested = 0;
+};
+
+// Scoped timer which adds a duration to a given externally reference timing duration
+class AdditiveScopedTimer
```

Comment:
Special case timer which adds to a given variable passed in.
Is scoped in case we throw exceptions during execution but also want the ability to explicit start/end the timer.